Quotient/connection.cpp

```diff
@@ -810,62 +806,49 @@ void Connection::requestDirectChat(const QString& userId)
     doInDirectChat(userId, [this](Room* r) { emit directChatAvailable(r); });
 }
 
-void Connection::requestDirectChat(User* u)
-{
-    doInDirectChat(u, [this](Room* r) { emit directChatAvailable(r); });
-}
-
 void Connection::doInDirectChat(const QString& userId,
                                 const std::function<void(Room*)>& operation)
 {
-    if (auto* u = user(userId))
-        doInDirectChat(u, operation);
-    else
-        qCCritical(MAIN)
-            << "Connection::doInDirectChat: Couldn't get a user object for"
-            << userId;
-}
+    auto* u = user(userId);
+    if (u == nullptr) {
+        return;
+    }
 
-void Connection::doInDirectChat(User* u,
-                                const std::function<void(Room*)>& operation)
-{
-    Q_ASSERT(u);
-    const auto& otherUserId = u->id();
     // There can be more than one DC; find the first valid (existing and
     // not left), and delete inexistent (forgotten?) ones along the way.
     DirectChatsMap removals;
     for (auto it = d->directChats.constFind(u);
          it != d->directChats.cend() && it.key() == u; ++it) {
         const auto& roomId = *it;
-        if (auto r = room(roomId, JoinState::Join)) {
-            Q_ASSERT(r->id() == roomId);
-            // A direct chat with yourself should only involve yourself :)
-            if (otherUserId == userId() && r->totalMemberCount() > 1)
-                continue;
-            qCDebug(MAIN) << "Requested direct chat with" << otherUserId
-                          << "is already available as" << r->id();
-            operation(r);
-            return;
-        }
-        if (auto ir = invitation(roomId)) {
-            Q_ASSERT(ir->id() == roomId);
-            auto j = joinRoom(ir->id());
-            connect(j, &BaseJob::success, this,
-                    [this, roomId, otherUserId, operation] {
-                        qCDebug(MAIN)
-                            << "Joined the already invited direct chat with"
-                            << otherUserId << "as" << roomId;
-                        operation(room(roomId, JoinState::Join));
-                    });
-            return;
-        }
-        // Avoid reusing previously left chats but don't remove them
-        // from direct chat maps, either.
-        if (room(roomId, JoinState::Leave))
+    if (auto r = room(roomId, JoinState::Join)) {
+        Q_ASSERT(r->id() == roomId);
+        // A direct chat with yourself should only involve yourself :)
+        if (userId == this->userId() && r->totalMemberCount() > 1)
```

Comment:
In a larger function like this I'd rather have the parameter named `otherUserId` (as the local variable used to be), otherwise it's rather easy to lose track which user id is involved.